Lower the `QUARKUS_CXF_MTOM_SOAK_ITERATIONS` default to 500 to make `MtomTest.soak()` pass on Quarkus Platform

For whatever reason the test is often failing on Quarkus Platform. 

```
2023-10-09T12:47:43.4398445Z 2023-10-09 12:47:43,413 INFO  [io.qua.cxf.it.ws.mto.ser.MtomTest] (main) Soaking with 64647 bytes, round 655
2023-10-09T12:47:43.4399382Z 2023-10-09 12:47:43,425 INFO  [io.qua.cxf.it.ws.mto.ser.MtomServiceImpl] (executor-thread-1) Received 64647 bytes of content type application/octet-stream
2023-10-09T12:51:43.5664660Z [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 277.001 s <<< FAILURE! - in io.quarkiverse.cxf.it.ws.mtom.server.MtomTest
2023-10-09T12:51:43.5673198Z [ERROR] io.quarkiverse.cxf.it.ws.mtom.server.MtomTest.soak  Time elapsed: 255.686 s  <<< ERROR!
2023-10-09T12:51:43.5674076Z java.lang.RuntimeException: java.net.SocketTimeoutException: Read timed out
2023-10-09T12:51:43.5675314Z 	at io.quarkiverse.cxf.it.ws.mtom.server.RandomBytesDataSource.count(RandomBytesDataSource.java:23)
2023-10-09T12:51:43.5676226Z 	at io.quarkiverse.cxf.it.ws.mtom.server.MtomTest.assertMtom(MtomTest.java:96)
2023-10-09T12:51:43.5677229Z 	at io.quarkiverse.cxf.it.ws.mtom.server.MtomTest.soak(MtomTest.java:57)
```

Let's lower the hard coded default in the test to 500 and let's keep `QUARKUS_CXF_MTOM_SOAK_ITERATIONS=1000` on our own CI.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -12,6 +12,10 @@ on:
       - 'LICENSE'
       - 'README*'
   pull_request:
+
+env:
+  QUARKUS_CXF_MTOM_LARGE_ATTACHMENT_INCREMENT_KB: 512
+
 concurrency:
   group: ${{ github.ref }}-${{ github.workflow }}
   cancel-in-progress: true

--- a/integration-tests/mtom/README.adoc
+++ b/integration-tests/mtom/README.adoc
@@ -1,9 +1,37 @@
 = Quarkus CXF MTOM tests
 
-The number of iterations performed by `io.quarkiverse.cxf.it.ws.mtom.server.MtomTest.soak()` can be set by setting
+== `MtomTest.soak()`
+
+The number of iterations performed by `io.quarkiverse.cxf.it.ws.mtom.server.MtomTest.soak()` can be set via
 
 [source,shell]
 ----
 $ export QUARKUS_CXF_MTOM_SOAK_ITERATIONS=100000
 $ mvn clean test -Dtest=MtomTest#soak
 ----
+
+The hard-coded default of 300 is intentionally low to make the test pass on Quarkus Platform CI.
+Higher values are more likely to uncover resource leaks.
+
+== `MtomTest.largeAttachment()`
+
+`MtomTest.largeAttachment()` can be tuned via
+
+[source,shell]
+----
+$ export QUARKUS_CXF_MTOM_LARGE_ATTACHMENT_INCREMENT_KB=512
+$ export QUARKUS_CXF_MTOM_LARGE_ATTACHMENT_MAX_KB=10240
+$ mvn clean test -Dtest=MtomTest#largeAttachment
+----
+
+The `QUARKUS_CXF_MTOM_LARGE_ATTACHMENT_INCREMENT_KB` is both the start size and increment at the same time (in KiB).
+The default of 1024 Kib (1 MiB) rather high for the test to pass quickly by default.
+Lower increments are more better for uncovering resource leaks or for pin-pointing protocol issues.
+
+`QUARKUS_CXF_MTOM_LARGE_ATTACHMENT_MAX_KB` has a default of 10240 KiB (10 MiB).
+The real max value of the attachment sent by the test is `QUARKUS_CXF_MTOM_LARGE_ATTACHMENT_MAX_KB` minus 889 bytes
+because Quarkus/Vert.x has a limit on the overall request size (configurable via `quarkus.http.limits.max-body-size`,
+default 10 MiB) and we need some space for the XML message and the multipart headers.
+
+Hence setting `QUARKUS_CXF_MTOM_LARGE_ATTACHMENT_MAX_KB` higher than 10240 KiB makes little sense without adjusting also
+`quarkus.http.limits.max-body-size`.

--- a/integration-tests/mtom/src/test/java/io/quarkiverse/cxf/it/ws/mtom/server/MtomTest.java
+++ b/integration-tests/mtom/src/test/java/io/quarkiverse/cxf/it/ws/mtom/server/MtomTest.java
@@ -28,7 +28,8 @@ public class MtomTest {
 
     /**
      * A reproducer for
-     * <a href="https://github.com/quarkiverse/quarkus-cxf/issues/973">https://github.com/quarkiverse/quarkus-cxf/issues/973</a>
+     * <a href=
+     * "https://github.com/quarkiverse/quarkus-cxf/issues/973">https://github.com/quarkiverse/quarkus-cxf/issues/973</a>
      *
      * @throws Exception
      */
@@ -36,7 +37,7 @@ public class MtomTest {
     public void soak() throws Exception {
         // The following fail with
         // Http2Exception: Flow control window exceeded for stream: 0
-        //   at io.netty.handler.codec.http2.Http2Exception.connectionError(Http2Exception.java:109)
+        // at io.netty.handler.codec.http2.Http2Exception.connectionError(Http2Exception.java:109)
         // final int size = 63 * KiB + 137; // fails at round 0
         // final int size = 63 * KiB + 136; // fails at round 5
         final int size = 63 * KiB + 135; // fails at round 5
@@ -48,9 +49,10 @@ public class MtomTest {
         // final int size = 63 * KiB + 129; // fails at round 5000
 
         // This one fails with a different exception:
-        // final int size = 63 * KiB + 128; // fails at round 10763 Corrupted channel by directly writing to native stream
+        // final int size = 63 * KiB + 128; // fails at round 10763 Corrupted channel by directly writing to native
+        // stream
         final int requestCount = Integer
-                .parseInt(Optional.ofNullable(System.getenv("QUARKUS_CXF_MTOM_SOAK_ITERATIONS")).orElse("1000"));
+                .parseInt(Optional.ofNullable(System.getenv("QUARKUS_CXF_MTOM_SOAK_ITERATIONS")).orElse("300"));
         log.infof("Performing %d interations", requestCount);
         for (int i = 0; i < requestCount; i++) {
             log.infof("Soaking with %d bytes, round %d", size, i);
@@ -60,19 +62,37 @@ public class MtomTest {
 
     /**
      * A reproducer for
-     * <a href="https://github.com/quarkiverse/quarkus-cxf/issues/973">https://github.com/quarkiverse/quarkus-cxf/issues/973</a>
+     * <a href=
+     * "https://github.com/quarkiverse/quarkus-cxf/issues/973">https://github.com/quarkiverse/quarkus-cxf/issues/973</a>
      *
      * @throws Exception
      */
     @Test
     public void largeAttachment() throws Exception {
 
-        int increment = 512 * KiB;
-        int maxSize = 10 * KiB * KiB - 512;
+        final int incrementKb = Integer
+                .parseInt(Optional.ofNullable(System.getenv("QUARKUS_CXF_MTOM_LARGE_ATTACHMENT_INCREMENT_KB")).orElse("1024"));
+        final int increment = incrementKb * KiB;
 
+        final int maxSizeKb = Integer
+                .parseInt(Optional.ofNullable(System.getenv("QUARKUS_CXF_MTOM_LARGE_ATTACHMENT_MAX_KB"))
+                        .orElse(String.valueOf(10 * KiB)));
+        log.infof("maxSizeKb %d ", maxSizeKb);
+        final int subtract = 889;
+        final int maxSize = (maxSizeKb * KiB) - subtract;
+
+        log.infof("Testing large attachments starting at %d KiB, incrementing %d KiB up to %d KiB - %d bytes", incrementKb,
+                incrementKb, maxSizeKb, subtract);
+        int lastSize = 0;
         for (int size = increment; size <= maxSize; size += increment) {
-            log.infof("Sending large attachment: %d bytes", size);
+            log.infof("Sending large attachment: %d bytes (%d KiB)", size, size / KiB);
             assertMtom(size);
+            lastSize = size;
+        }
+        if (maxSize > lastSize) {
+            /* Make sure that we really test the max size */
+            log.infof("Sending max sized attachment: %d bytes (%d KiB)", maxSize, maxSize / KiB);
+            assertMtom(maxSize);
         }
 
     }

